### DISCUSSION TITLE
Suggestion: NTP pool vendor zone

### DISCUSSION THREAD

**Discussion by rimrul** [Hypfer/Valetudo#1978] [Q&A / Support] [answered]
Valetudo currently defaults to `pool.ntp.org` in its NTP config. The operators of `pool.ntp.org` don't like when software/hardware/os vendors do that and [ask that they instead request and use a so called vendor zone](https://www.ntppool.org/en/vendors.html) (a group of hostnames designated to them, for example `0.valetudo.pool.ntp.org`, `1.valetudo.pool.ntp.org`, ...). Vendor zones are free for open source projects, so we should probably try to work with them.

**Comment by stefanmd023** [comment]
Keep in mind that most users of Valetudo will prefer to use their own local NTP server to keep the robot off the internet

**Reply by rimrul** [reply]
Do we have any statistics about that or is this based on assumptions?

**Reply by stefanmd023** [reply]
There are no statistics because that goes against the spirit of the project, which is to use your appliance completely offline, hence the majority will actually have to setup a local NTP server to have it offline AND keep schedules, because they are that type of people, and this attitude is encouraged by the Devs and the community.

**Comment by Hypfer** [comment, marked as answer]
~~Interesting input, but honestly I don't think that Valetudo qualifies for such a special treatment given just how niche it is.~~

~~Looking at the examples given in their docs:~~
~~- Appliance vendors (Netgear, D-Link, Linksys, ...)~~
~~- Operating System vendors (Debian, RedHat, FreeBSD, m0n0wall, ...)~~

~~I think it's safe to assume that that feature is for fleets of tens to hundreds of thousands of devices. I don't think that Valetudo has even reached 10k devices. And those Valetudo devices that do exist make a single query once every 8 hours randomized by Valetudo startup time.
Of course it will grow but to me it still doesn't seem even remotely similar to the other things mentioned in their docs.~~

~~I also don't like how that would add some kind of infrastructure that would have to be maintained. Even though it's not actual servers, because of the special treatment, I'd have to be available for the NTP Pool team to contact and manage this stuff. That requires a commitment to do so in a timely fashion and thus is a liability. This also _has_ to be done by me because this is my project of which I'm in charge. It wouldn't make sense to have someone else be in charge of that.~~

~~That said, if the Team behind the NTP Pool project wants it to be done that way then of course it will be done that way.
It's their project and their free service so they get to decide the rules behind that. To me, it just doesn't look like that the Valetudo use-case is actually what they're talking about in that docs. But then again I might be wrong so feel free to correct me.~~

~~I just wouldn't want to proactively start this if it's not required because it's just more work, more things that can break and more stuff that has to be maintained for no reason. That also includes asking if it should be done so please don't do that.
If the topic becomes relevant, I'm quite certain that someone will reach out on their own. No need to force that happening sooner.~~

**Reply by Hypfer** [reply]
> Open source projects
> Open Source projects are of course particularly welcome to use the pool in their default setup, but we ask that you [get a vendor zone](https://www.ntppool.org/en/vendors.html#vendor-zone) when using the pool as a default configuration.

Huh. As it turns out, further down that page there is this paragraph which is indeed quite clear though easily missable.
Well nvm then. Time to apply for a vendor zone I suppose

**Reply by rimrul** [reply]
> Interesting input, but honestly I don't think that Valetudo qualifies for such a special treatment given just how niche it is.
> 
> Looking at the examples given in their docs:
> - Appliance vendors (Netgear, D-Link, Linksys, ...)
> - Operating System vendors (Debian, RedHat, FreeBSD, m0n0wall, ...)

They do also specifically mention other software that does use ntp.
 
>
> I think it's safe to assume that that feature is for fleets of tens to hundreds of thousands of devices. I don't think that Valetudo has even reached 10k devices. 

The registration form for a zone has smaller increments for estimated number of devices:

> <500
> <2500
> <5000
> <10000


> And those Valetudo devices that do exist make a single query once every 8 hours.

Their recommende query frequency ranges from once a day to a few times per hour, so that's not exactly exceptionally rare (but of course a good default value). 

> Of course it will grow but to me it still doesn't seem even remotely similar to the other things mentioned in their docs.
> 
> I also don't like how that would add some kind of infrastructure that would have to be maintained. Even though it's not actual servers, because of the special treatment, I'd have to be available for the NTP Pool team to contact and manage this stuff. That requires a commitment to do so in a timely fashion and thus is a liability.

fair.

> That said, if the Team behind the NTP Pool project wants it to be done that way then of course it will be done that way.
> It's their project and their free service so they get to decide the rules behind that.

Well, their written rules about this are

> Open Source projects are of course particularly welcome to use the pool in their default setup, but we ask that you get a vendor zone when using the pool as a default configuration. 	

and 

> Do not use the standard pool.ntp.org names as a default configuration in your system. 

And

> For internal use on systems you manage you do not need a vendor zone. For initial development or small-scale testing of an appliance or software, you don't either. However, you must setup a vendor zone before production deployments or distribution start. 

I guess you could consider Valetudo "small scale testing" if you want to.

> I just wouldn't want to proactively start this if it's not required because it's just more work, more things that can break and more stuff that has to be maintained for no reason. That also includes asking if it should be done so please don't do that.

I've brought this to your attention, you decide about wether anything needs to be done about this. Either way is fine by me.

> If the topic becomes relevant, I'm quite certain that someone will reach out on their own. No need to force that happening sooner.

**Reply by Hypfer** [reply]
Once you're actually logged in into the ntppool portal and navigate to the vendor page, it's also quite clear:

> If you are distributing an appliance, operating system or other kind of software using the NTP Pool, you need to setup a "[vendor zone](https://www.ntppool.org%20/vendors.html)".
>For internal use on systems you manage you do not need a vendor zone. For initial development or small-scale testing of an appliance or software, you don't either. However, you must setup a vendor zone before production deployments or distribution start.
>For open source projects this is free; for commercial products, including commercial open source, we ask for a nomimal annual fee (based on the number of clients) to support the NTP Pool Project infrastructure.

But to see this, you first need to end up there with an active login session and for that you need to know that you're supposed to end up there, because why else would you have an account (besides running your own server and adding it to the pool).
So.. eh.

Thanks for bringing this up @rimrul :)

I think it would be helpful if their documentation would contain a bit more information on _why_ they want projects to use a vendor zone. From what I initially saw (and given the ticket process) it looked like something they want less of but _if_ it is about gaining insight into which projects use how many resources (and thus enable them to act on it) they'd actually want _more_ vendor zones.

This is all unclear from the docs and "they don't like this" from a third party - even though you've earned your credibility - is easy to dismiss as the usual noise in FOSS

**Reply by Hypfer** [reply]
The vendor zone just got approved.
Done
https://github.com/Hypfer/Valetudo/commit/1a3a9f8b81eb923a9d7ee7858db9acab49aa88d2

The next release will migrate existing instances to it
